mod-users-bl/src/main/resources/domain/mod-users-bl/features/users-bl.feature

```diff
@@ -77,3 +74,89 @@ Feature: Test user business logic
     When method POST
     Then status 201
     And match responseHeaders contains { 'x-okapi-token': '#present' }
+
+  Scenario: Return a composite object for the currently logged in user
+
+    Given path 'bl-users/_self'
+    When method GET
+    Then status 200
+    And match response.user.active == true
+    And match response.user.personal.firstName == 'Admin'
+    And match response.user.username == 'test-admin'
+
+  Scenario: Return a composite object for the currently logged in user by username
+
+    Given path 'bl-users/by-username/' + 'test-admin'
+    When method GET
+    Then status 200
+    And match response.user.active == true
+    And match response.user.personal.firstName == 'Admin'
+    And match response.user.username == 'test-admin'
+
+  Scenario: Return a composite object referenced by the user's id
+
+    Given path 'bl-users/by-id/' + '00000000-1111-5555-9999-999999999991'
+    When method GET
+    Then status 200
+    And match response.user.active == true
+    And match response.user.personal.firstName == 'Admin'
+    And match response.user.username == 'test-admin'
+
+  Scenario: Return an object listing number of open transactions that are associated to the user referenced by the user's username
+    # creating owner & fine is only for opening & testing a transaction for testAdmin user.
+    Given path 'owners'
+    And request
+    """
+    {
+     "owner": "testOwner",
+     "desc": "for Testing",
+     "id": "6312d172-f0cf-40f6-b27d-9fa8feaf332f"
+    }
+    """
+    When method POST
+    Then status 201
+
+    Given path 'accounts'
+    And request
+    """
+    {
+     "ownerId": "6312d172-f0cf-40f6-b27d-9fa8feaf332f",
+     "feeFineId": "6312d173-f0cf-40f6-b27d-9fa8feaf332f",
+     "amount": 15,
+     "paymentStatus": {
+     "name": "Outstanding"
+     },
+     "status": {
+     "name": "Open"
+     },
+     "remaining": 5,
+     "feeFineType": "printing card",
+     "feeFineOwner": "owner",
+     "userId": "00000000-1111-5555-9999-999999999991",
+     "id": "6312d172-f0cf-40f6-b27d-9fa8feaf333f"
+    }
+    """
+    When method POST
+    Then status 201
+
+    Given path 'bl-users/by-username/test-admin/open-transactions'
+    When method GET
+    Then status 200
+    And match response.hasOpenTransactions == true
+    And match response.feesFines == 1
+
+  Scenario: Return an object listing number of open transactions that are associated to the user referenced by the user's id
+
+    Given path 'bl-users/by-id/00000000-1111-5555-9999-999999999991/open-transactions'
+    When method GET
+    Then status 200
+    And match response.hasOpenTransactions == true
+    And match response.feesFines == 1
+
+  Scenario: Delete a user referenced by user's id after checking for open transactions. Deletion will be executed if and only if the user has no open transactions.
```

Comment:
I think it is better to describe the intended behaviour of the test in the name of the scenario

```suggestion
  Scenario: Disallow deletion of user with open transactions
```

Reply:
Yes , i will change it . thankyou